sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/Channel.java

```diff
@@ -10,6 +10,14 @@
  */
 public interface Channel {
 
+    /**
+     * Set app secret. Intended usage is to use that only if there was no app secret at initialization time.
+     * The behavior is undefined if trying to update app secret a second time.
```

Comment:
Why would the behavior be undefined?

Reply:
Because we don't try to handle that case since it does not happen. Undefined as not defined by the spec. We validate inputs and flow in AppCenter already (configure no more than once).

Reply:
(this is an internal class not meant for public usage)